### PR TITLE
Update DeckPicker.java(Corrected the multiple changes issue)

### LINKED ISSUE #9961: [Bug] Login does not have "try anyway" feature if user is detected as being offline
###### Reproduction Steps

1. Block google from a firewall so that your android device will report that you are not connected to internet, while you still have access to the anki server.
2. Try to connect to anki.

###### Expected Result

Either attempt to connect before and then report that there is no connectivity on failure, or present the "Try anyway" button as it happens for syncronization.

###### Actual Result

AnkiDroid says "You are offline" and doesn't allow to connect to the server at all.

###### Research

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by SSTG**
## Purpose / Description
_Describe the problem or feature and motivation_

## Fixes
Fixes #9961 

## Approach
_How does this change address the problem?_

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration (SDK version(s), emulator or physical, etc)

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [ ] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [ ] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [ ] You have commented your code, particularly in hard-to-understand areas
- [ ] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java [hunk 1]
```diff
@@ -1945,7 +1946,8 @@ public void onPostExecute(Payload data) {
                             showSyncErrorMessage(joinSyncMessages(dialogMessage, syncMessage));
                             break;
                         case CONNECTION_ERROR:
-                            dialogMessage = res.getString(R.string.sync_connection_error);
+                            Connection.isOnline();
```
Comment (legarspol) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java:1949]: - [ ] My understanding is that this method has no side effects and you don't use the return value. So calling it has no effects hence it should be removed.
- [ ] The tests won't pass until you delete the sync_connection_error resource string, which is now unused. Look at the tests this PR initiated.

**Comment by ShridharGoel** [comment]
@SSTG Can you fill the PR template? Also, it's important to link the issue in the PR description so that when this PR is merged, the linked issue automatically gets closed.

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java (hunk 1 above)
Comment (SSTG) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java]: Oh I didn't realise all this. I will correct them all Thanks

Comment (SSTG) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java]: I have updated the code just as you mentioned. But I cannot find this resource string anywhere. I have been looking for it for a while now

**Review by mikehardy — CHANGES_REQUESTED** [review]
Hi there! Thanks for taking this on

We must not hard code strings, we support lots and lots of languages and need to use the translation infrastructure: https://github.com/ankidroid/Anki-Android/wiki/Development-Guide#handling-translations

There is extra space (an extra line) which is unnecessary for the fix

As @ShridharGoel mentioned, you must fill out the PR template, you do link to the original issue now, but I don't see a description of how this PR affects that issue. Right now I'm not exactly sure how this change affects the related issue?

**Comment by mikehardy** [comment]
Superceded by #10414

**Pull request closed without merging**